src/packages/@ncigdc/components/analysis/ClinicalAnalysisResult.js

```diff
@@ -463,4 +518,10 @@ const ClinicalAnalysisResult = ({
   );
 };
 
-export default enhance(ClinicalAnalysisResult);
+const ClinicalAnalysisResultWithSurvivalData = lifecycle({
```

Comment:
this can move to the enhance function, after you've defined the method in `withProps`. I'm also wondering if we can call `populateSurvivalData` using `withPropsOnChange` and checking when the `setId` from currentAnalysis changes, and skip the lifecycle methods. Otherwise we can use `componentDidMount` and `componentDidUpdate`

Reply:
> call populateSurvivalData using withPropsOnChange

that worked, and fixed the issue with survival plots not updating when you change to another set of data :+1: